File: src/yl13/GameCoreAPI/EventListener.php
<?php

declare(strict_types=1);

namespace yl13\GameCoreAPI;

use pocketmine\event\Listener;

use pocketmine\event\player\{
    PlayerJoinEvent, PlayerQuitEvent, PlayerChatEvent
};

class EventListener implements Listener {

    private $plugin;

    public function __construct(GameCoreAPI $plugin) {
        $this->plugin = $plugin;
    }

    public function onPlayerJoin(PlayerJoinEvent $ev) {
        $player = $ev->getPlayer();
        $this->plugin->initPlayerData($this->plugin, $player);
        if(is_string($this->plugin->getConfigure('chatchannel', 'default'))) {
            //让玩家加入默认聊天频道如果有的话
            $ChatChannel = $this->plugin->get($this->plugin, 'CHATCHANNEL');
            $ChatChannel[$this->plugin->getConfigure('chatchannel', 'default')]['players'][$player->getName()] = $player;
            $this->plugin->set($this->plugin, 'CHATCHANNEL', $ChatChannel);
            $this->plugin->setPlayerData($this->plugin, $player, 'CHATCHANNEL', $this->plugin->getConfigure('chatchannel', 'default'));
        }
    }

    public function onPlayerQuit(PlayerQuitEvent $ev) {
        $player = $ev->getPlayer();
        $ChatChannel = $this->plugin->get($this->plugin, 'CHATCHANNEL');
        if(isset($ChatChannel[$this->plugin->getConfigure('chatchannel', 'default')]['players'][$player->getName()])) {
            unset($ChatChannel[$this->plugin->getConfigure('chatchannel', 'default')]['players'][$player->getName()]);
        }
        //TODO: 储存金钱数据
        $this->plugin->removePlayerData($this->plugin, $player);
    }

    public function onPlayerChat(PlayerChatEvent $ev) {
        if($this->plugin->getConfigure('chatchannel', 'enabled')) {
            $ev->setCancelled(true);
            $player = $ev->getPlayer();
            $ChatChannel = $this->plugin->get($this->plugin, 'CHATCHANNEL');
            $playerData = $this->plugin->getPlayerData($this->plugin, $player);
            if(isset($ChatChannel[$playerData['chatchannel']])) {
                $players = $ChatChannel[$playerData['chatchannel']]['players'];
                $format = $ChatChannel[$playerData['chatchannel']]['format'];
                if(!$ChatChannel[$playerData['chatchannel']]['mute']) {
                    if($format != null) {
                        $format = str_replace(['PLAYER_NAME', 'MESSAGE'], [$player->getName(), $ev->getMessage()], $format);
                        $this->plugin->getServer()->broadcastMessage($format, $players);
                    }
                    $this->plugin->getServer()->broadcastMessage("[{$player->getName()}] {$ev->getMessage()}", $players);
                }
            }
        }
    }
}

File: src/main/java/xyz/hjdmc/gamecoreapi/event/EventListener.java
package xyz.hjdmc.gamecoreapi.event;

import cn.nukkit.Player;
import cn.nukkit.event.EventHandler;
import cn.nukkit.event.EventPriority;
import cn.nukkit.event.Listener;
import cn.nukkit.event.player.PlayerChatEvent;
import cn.nukkit.event.player.PlayerJoinEvent;
import cn.nukkit.event.player.PlayerQuitEvent;
import xyz.hjdmc.gamecoreapi.GameCoreAPI;
import xyz.hjdmc.gamecoreapi.utils.ChatChannel;
import xyz.hjdmc.gamecoreapi.utils.InGamePlayer;

import java.util.ArrayList;

public class EventListener implements Listener {

    private final GameCoreAPI plugin;

    public EventListener(GameCoreAPI plugin) {
        this.plugin = plugin;
    }

    @EventHandler(priority = EventPriority.NORMAL, ignoreCancelled = false)
    public void onPlayerJoin(PlayerJoinEvent event) {
        this.plugin.addInGamePlayer(this.plugin, new InGamePlayer(this.plugin, event.getPlayer()));
    }

    @EventHandler(priority = EventPriority.NORMAL, ignoreCancelled = false)
    public void onPlayerQuit(PlayerQuitEvent event) {
        this.plugin.removeInGamePlayer(this.plugin, event.getPlayer());
    }

    @EventHandler(priority = EventPriority.NORMAL, ignoreCancelled = false)
    public void onPlayerChat(PlayerChatEvent event) {
        event.setCancelled();
        if(event.getPlayer().isOp()) {
            this.plugin.getServer().broadcastMessage("[" + event.getPlayer().getName() + "] " + event.getMessage());
        }
        ChatChannel chatChannel = this.plugin.getInGamePlayer(this.plugin, event.getPlayer()).getChatChannel();
        if(chatChannel != null) {
            Player[] players = chatChannel.getAllPlayer();
            String format = chatChannel.getFormat();
            if(!format.equals("")) {
                format = format.replace("MESSAGE", event.getMessage());
                format = format.replace("PLAYERNAME", event.getPlayer().getName());
                this.plugin.getServer().broadcastMessage(format, players);
            } else {
                this.plugin.getServer().broadcastMessage(event.getPlayer().getName() + ": " + event.getMessage());
            }
        }
    }
}
